snakes_battle/snake.py

```diff
@@ -118,7 +125,23 @@ def shrink(self, shrinking_amount):
         self.body_pos = self.body_pos[:self.length] # Removing nodes from the snake
 
     def eat(self, fruit):
```

Comment:
Why did you return all the fruit action implementation to the snake?
In my opinion this is really not elegant.. All these if else, and the concept of adding a fruit should change the snake's code is not good, and also makes the code very clumsy.